packages/office-ui-fabric-react/src/components/List/List.tsx

```diff
@@ -318,10 +318,13 @@ export class List extends BaseComponent<IListProps, IListState> {
 
     for (let i = 0; page.items && i < page.items.length; i++) {
       let item = page.items[i];
-      let itemKey = (item ? item.key : null);
+      const defaultKey = page.startIndex + i;
+      let itemKey =
+        item &&
+        (this.props.getKey ? this.props.getKey(item, defaultKey) : item.key);
```

Comment:
if you have `getKey` and you don't have `item`, do you still want to call `getKey` with the index? Otherwise `onRenderCell` will be called for a null/undefined item but not `getKey` (if passed)